jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/GltfLoader.java

```diff
@@ -464,6 +464,10 @@ public Geometry[] readMeshPrimitives(int meshIndex) throws IOException {
             // Read mesh extras
             mesh = customContentManager.readExtensionAndExtras("primitive", meshObject, mesh);
             Geometry geom = new Geometry(null, mesh);
+            // Cached data about the geometry needed by TextureTransformExtensionLoader.class
+            addToCache("lastCreatedGeom", 0, geom, 3); // Last created geometry
+            addToCache("lastCreatedGeom", 1, null, 3); // Last processed geometry
+            addToCache("lastCreatedGeom", 2, null, 3); // Last applied transformation matrix (inverted)
```

Comment:
These should be removed. GltfLoader should know nothing about TextureTransformExtensionLoader.

Instead, you should grab the mesh in TextureTransformExtensionLoader

```
public Object handleExtension(GltfLoader loader, String parentName, JsonElement parent, JsonElement extension, Object input) throws IOException {
        if (input instanceof Mesh) {
            loader.addToCache("textureTransformData", 0, mesh, 2); // Last created mesh
            loader.addToCache("textureTransformData", 1, null, 2); // Inverted transform matrix
        } else if (input instanceof Texture2D) {
             Mesh mesh = loader.fetchFromCache("textureTransformData", 0, mesh, Mesh.class);
             // do things...
        }
}

```

you do not need  "Last processed geometry", just check with the transformation matrix, if it is null then calculate and apply it and if it is not null then it means this is from the next texture of the same material, you need to calculate matrix again and check with the previous one as you have done in `if(!m.mult(mInvLast).isIdentity())` and warn if they are different.